lib/DiscoveryStreamFeed.jsm

```diff
@@ -189,12 +203,13 @@ this.DiscoveryStreamFeed = class DiscoveryStreamFeed {
   }
 
   async _checkExpirationPerComponent() {
-    const cachedData = await this.cache.get() || {};
-    const {feeds} = cachedData;
+    const layout = await this.kvstoreGet("layout") || {};
```

Comment:
How many key/value pairs do we have in this store? If there are only three of them we can just load all of them in one call using the [enumerator](https://searchfox.org/mozilla-central/source/toolkit/components/kvstore/kvstore.jsm#173).